cluster-install/install.sh

```diff
@@ -1708,22 +1886,48 @@ step_check_volumesnapshotclasses() {
-            add_problem "namespace '$NAMESPACE': not found but IMAGE_PULL_SECRET is set" "$err_msg"
+step_check_namespace_and_pull_secret_exist() {
+    local -r namespace="$NAMESPACE"
+    local -r pull_secret="$IMAGE_PULL_SECRET"
+    local err_msg=""
+
+    [ -z "$namespace" ] && return 0
+
+    # Best effort guess at which registry to put in the create command
+    local registry="<REGISTRY>"
+    if components_include_connector; then
+        registry="$CONNECTOR_IMAGE_REGISTRY"
+    # Skip if Trident's image is still set to our default of 'docker.io/netapp', which is public
+    elif components_include_trident && ! str_contains_at_least_one "$(get_config_trident_image)" "docker.io/netapp" ; then
+        registry="$TRIDENT_IMAGE_REGISTRY"
+    elif components_include_acp; then
+        registry="$TRIDENT_ACP_IMAGE_REGISTRY"
+    fi
+
+    local create_secret_cmd="kubectl create secret docker-registry '$pull_secret' -n '$namespace'"
+    create_secret_cmd+=" --docker-server='$registry' --docker-username='<USERNAME>' --docker-password='<PASSWORD>'"
+
+    if ! k8s_resource_exists "namespace/$namespace"; then
+        if [ -n "$pull_secret" ]; then
+            err_msg="The specified NAMESPACE '$namespace' does not exist on the cluster, but IMAGE_PULL_SECRET is set."
+            err_msg+=" Please create the namespace and secret:"
+            err_msg+="${__NEWLINE}    - kubectl create namespace '$namespace'"
+            err_msg+="${__NEWLINE}    - $create_secret_cmd"
+            add_problem "namespace '$namespace': not found but IMAGE_PULL_SECRET is set" "$err_msg"
             exit_if_problems
         fi
-        if prompt_user_yes_no "The namespace $NAMESPACE doesn't exist. Create it now? Choosing 'no' will exit"; then
-            kubectl create namespace "$NAMESPACE" --dry-run=client -o yaml | kubectl apply -f -
-            logdebug "namespace '$NAMESPACE': OK"
+        if prompt_user_yes_no "The namespace $namespace doesn't exist. Create it now? Choosing 'no' will exit"; then
+            kubectl create namespace "$namespace" --dry-run=client -o yaml | kubectl apply -f -
+            logdebug "namespace '$namespace': OK"
         else
-            local -r err_msg="User chose not to create the namespace. Please create the namespace and/or run the script again."
+            err_msg="User chose not to create the namespace. Please create the namespace and/or run the script again."
             add_problem "User chose not to create the namespace. Exiting." "$err_msg"
             exit_if_problems
         fi
+    elif [ -n "$pull_secret" ] && ! k8s_resource_exists "secret/$pull_secret" "$namespace"; then
+        err_msg="The specified IMAGE_PULL_SECRET '$pull_secret' does not exist in namespace '$namespace'."
+        err_msg+=" Please create it:${__NEWLINE}    - $create_secret_cmd"
+        add_problem "pull secret '$pull_secret' not found in namespace '$namespace'" "$err_msg"
```

Comment:
This is specifically if the namespace exists, but the pull secret doesn't. We also print the generated pull secret command (but not the namespace one).